nilearn/decomposition/dict_learning.py

```diff
@@ -0,0 +1,202 @@
+        path to the caching directory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs: integer, optional
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    verbose: integer, optional
+        Indicate the level of verbosity. By default, nothing is printed
+
+    References
+    ----------
+    * G. Varoquaux et al. "A group model for stable multi-subject ICA on
+      fMRI datasets", NeuroImage Vol 51 (2010), p. 288-299
+
+    * G. Varoquaux et al. "ICA-based sparse features recovery from fMRI
+      datasets", IEEE ISBI 2010, p. 1177
+    """
+
+    def __init__(self, mask=None, n_components=20,
+                 smoothing_fwhm=6, do_cca=True,
+                 threshold='auto', n_init=10,
+                 standardize=True,
+                 random_state=0,
+                 target_affine=None, target_shape=None,
+                 low_pass=None, high_pass=None, t_r=None,
+                 alpha=1,
+                 batch_size=10,
+                 n_iter=1000,
+                 reduction=False,
+                 # Common options
+                 memory=Memory(cachedir=None), memory_level=0,
+                 n_jobs=1, verbose=0,
+                 ):
+        CanICA.__init__(self,
+                        mask=mask, memory=memory, memory_level=memory_level,
+                        n_jobs=n_jobs, verbose=max(0, verbose - 1), do_cca=do_cca,
+                        threshold=threshold, n_init=n_init,
+                        n_components=n_components, smoothing_fwhm=smoothing_fwhm,
+                        target_affine=target_affine, target_shape=target_shape,
+                        random_state=random_state, high_pass=high_pass, low_pass=low_pass,
+                        t_r=t_r,
+                        keep_data_mem=True,
+                        standardize=standardize)
+        self.reduction = reduction
+        # Setting n_jobs = 1 as it is slower otherwise
+        MiniBatchDictionaryLearning.__init__(self, n_components=n_components, alpha=alpha,
+                                             n_iter=n_iter, batch_size=batch_size,
+                                             fit_algorithm='lars',
+                                             transform_algorithm='lasso_lars',
+                                             transform_alpha=alpha,
+                                             verbose=max(0, verbose - 1),
+                                             random_state=random_state,
+                                             shuffle=True,
+                                             n_jobs=1)
+
+    def _init_dict(self, imgs, y=None, confounds=None):
+        CanICA.fit(self, imgs, y=y, confounds=confounds)
+        self.data_flat_ = np.concatenate(self.data_flat_, axis=0)
```

Comment:
not sure of the context, but if data_flat is a list, you need to
concatenate. If it is an array of unknown ordering, you should ravel. The
fact that ravel was fast points to the fact that it was C ordered, in which
case the most explicit thing to do is self.data_flat.shape = -1,



On Thursday, July 9, 2015, Danilo Bzdok <notifications@github.com> wrote:

> In nilearn/decomposition/dict_learning.py
> <https://github.com/nilearn/nilearn/pull/638#discussion_r34205287>:
>
> > +                        standardize=standardize)
> > +        self.reduction = reduction
> > +        # Setting n_jobs = 1 as it is slower otherwise
> > +        MiniBatchDictionaryLearning.__init__(self, n_components=n_components, alpha=alpha,
> > +                                             n_iter=n_iter, batch_size=batch_size,
> > +                                             fit_algorithm='lars',
> > +                                             transform_algorithm='lasso_lars',
> > +                                             transform_alpha=alpha,
> > +                                             verbose=max(0, verbose - 1),
> > +                                             random_state=random_state,
> > +                                             shuffle=True,
> > +                                             n_jobs=1)
> > +
> > +    def _init_dict(self, imgs, y=None, confounds=None):
> > +        CanICA.fit(self, imgs, y=y, confounds=confounds)
> > +        self.data_flat_ = np.concatenate(self.data_flat_, axis=0)
>
> In my little timeit-experiment, np.ravel() was faster.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/638/files#r34205287>.
>